### PR TITLE
[BOB-380] Add IValidateable interface

### PR DESCRIPTION
**Pull request by RJSonnenberg**
### Description

**NOTE: It will be easier to review this PR by stepping through each commit.**

- Update Documentation (first commit)
  - Correct typos, grammar, examples, formatting
  - Reword for clarity
  - Add more clarity to some sections
- Add `IValidateable<'F>` interface type and documentation (second commit)
- Add example for using `IValidateable` interface in concert with Giraffe model validation (third commit)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1568,6 +1576,175 @@ In this example, the `NewUser` type implements the `IValidateable<NewUser>` inte
...
+In this case, the failure type of the `IValidateable<'F>` represented by `'F` is `NewUserFailures`.
+
+```fsharp
+[<CLIMutable>]
+type NewUserVM =
+    { Name: string option
+      Username: string option
+      Password: string option }
+
+    interface IValidateable<NewUserFailures> with
```
Comment (devinlyons) [inline README.md:1651]: I don't understand the point of the interface. What's it for?

File: README.md [hunk 2]
```diff
@@ -1568,6 +1576,175 @@ In this example, the `NewUser` type implements the `IValidateable<NewUser>` inte
...
+
+Now, we need to implement the Giraffe `IModelValidation<'T>` interface and its `Validate` method.
+This is what will apply the validations and throw an appropriate HTTP response.
+
+```fsharp
+[<CLIMutable>]
+type NewUserVM =
+    // ... nothing new here
+
+    interface IModelValidation<NewUserVM> with
```
Comment (devinlyons) [inline README.md:1698]: This implementation suggests validation without transformation. That's antithetical to the design of this library.

File: README.md [hunk 3]
```diff
@@ -1568,6 +1576,175 @@ In this example, the `NewUser` type implements the `IValidateable<NewUser>` inte
...
+        member this.Validate() =
+            match
+                // To access Validate() on the IValidateable<NewUserFailures> interface,
+                // we must upcast to that interface.
+                (this :> IValidateable<NewUserFailures>).Validate()
```
Comment (devinlyons) [inline README.md:1703]: The IValidatable method doesn't really do anything here that a simple function call couldn't handle.

**Comment by RJSonnenberg** [comment]
Closing this PR in favor of a separate Giraffe model validation interface #17

**Pull request closed without merging**